### ISSUE TITLE
Rust-analyzer shows an error for code that compiles

### ISSUE THREAD

**Issue by dirkvdb** [rust-lang/rust-analyzer#20662]
Rust analyzer version `0.3.2593-standalone (6b2e677795 2025-08-25)`
rustc version  `1.89.0 (29483883e 2025-08-04)`
editor extension: both vscode and zed show the same errors

For a few months I'm getting a few lines of code that are flagged as an error by rust-analyzer but the code compiles just fine with rustc. I tried to simplify the code to a minimal snippet that still triggers the error. The error reported by rust-analyzer is: `cannot mutate immutable variable 'b' [E0384]`

**code snippet to reproduce**:
```rust
pub trait A: std::ops::IndexMut<usize> {
    type T: A;
}

fn func(a: &mut impl A, b: &mut [i32]) {
    b[0] += 1;
}
```

Removing the unused argument `a` causes the error to go away.

**Comment by lnicola** [comment]
Saw this on my previous version, but I can't reproduce it on `master`:

<img width="544" height="459" alt="Image" src="https://github.com/user-attachments/assets/e77b83f2-7a97-42fc-b52b-aeb9cce610b4" />

**Comment by ChayimFriedman2** [comment]
So it's fixed? https://github.com/rust-lang/rust-analyzer/pull/20654 could very well fix this. What's weird is that OP isn't using an unstable version of r-a.

**Comment by lnicola** [comment]
I think it predates the next solver, I just tested on the latest stable version and it's there.

**Comment by ShoyuVanilla** [comment]
> I think it predates the next solver, I just tested on the latest stable version and it's there.

Yes, I can reproduce this with stable releases.

**Comment by jsonDoge** [comment]
Same, showed old issues, reverting back to `0.3.2577` fixed it for me

**Comment by ChayimFriedman2** [comment]
Can't reproduce on master, so closing.

**Comment by lnicola** [comment]
Might be worth adding a test.

**Comment by ChayimFriedman2** [comment]
Done: https://github.com/rust-lang/rust-analyzer/pull/20802
